From d73e3464a2f3f5d66f07d08a5e6aab8a63729e46 Mon Sep 17 00:00:00 2001
From: Ricardo Quesada <ricardoquesada@gmail.com>
Date: Tue, 29 Aug 2023 20:02:18 -0700
Subject: [PATCH] HFP_AG: Increate buffer for snprintf

The required buffer should be 22 instead of 20.
Prevents possible buffer overflow.

Fixes issues: https://github.com/bluekitchen/btstack/issues/523
---
 src/classic/hfp_ag.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/classic/hfp_ag.c b/src/classic/hfp_ag.c
index 3d0445c36..4a1b03cc2 100644
--- a/src/classic/hfp_ag.c
+++ b/src/classic/hfp_ag.c
@@ -206,7 +206,7 @@ static int has_hf_indicators_feature(hfp_connection_t * hfp_connection){
 /* unsolicited responses */
 
 static int hfp_ag_send_change_in_band_ring_tone_setting_cmd(hfp_connection_t * hfp_connection){
-    char buffer[20];
+    char buffer[22];
     snprintf(buffer, sizeof(buffer), "\r\n%s: %d\r\n",
              HFP_CHANGE_IN_BAND_RING_TONE_SETTING, use_in_band_tone(hfp_connection));
     buffer[sizeof(buffer) - 1] = 0;
-- 
2.41.0

